### ISSUE TITLE
setup.py exit code 2, installing from pip into venv

### ISSUE THREAD

**Issue by Fincap** [PyWavelets/pywt#549]
Hi,

I'm running Python 64-bit 3.8.2 on a Windows 10 machine. I'm attempting to install PyWavelets (dependency on another tool I'm trying to run) into a virtual environment using pip.

My only lead is that this might be an issue with my install of VS Build Tools (which I have reinstalled since getting this error, which has not fixed the error).

I get the following:
```
Running setup.py install for PyWavelets ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\finnk\pycharmprojects\onkodicom\venv\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\finnk\\AppData\\Local\\Temp\\pip-install-nc9fzkly\\PyWavelets\\setup.py'"'"'; __file__='"'"'C:\\Users\\finnk\\AppData\\Local\\Temp\\pip-install-nc9fzkly\\PyWavelets\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\finnk\AppData\Local\Temp\pip-record-g0a3zlnv\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\finnk\pycharmprojects\onkodicom\venv\include\site\python3.8\PyWavelets'
         cwd: C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\
    Complete output (150 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.8
    creating build\lib.win-amd64-3.8\pywt
    copying pywt\version.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\_cwt.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\_dwt.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\_functions.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\_multidim.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\_multilevel.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\_swt.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\_thresholding.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\_wavelet_packets.py -> build\lib.win-amd64-3.8\pywt
    copying pywt\__init__.py -> build\lib.win-amd64-3.8\pywt
    creating build\lib.win-amd64-3.8\pywt\_extensions
    copying pywt\_extensions\__init__.py -> build\lib.win-amd64-3.8\pywt\_extensions
    creating build\lib.win-amd64-3.8\pywt\data
    copying pywt\data\create_dat.py -> build\lib.win-amd64-3.8\pywt\data
    copying pywt\data\_readers.py -> build\lib.win-amd64-3.8\pywt\data
    copying pywt\data\__init__.py -> build\lib.win-amd64-3.8\pywt\data
    creating build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_concurrent.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_cwt_wavelets.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_data.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_deprecations.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_doc.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_dwt_idwt.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_functions.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_matlab_compatibility.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_matlab_compatibility_cwt.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_modes.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_multidim.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_multilevel.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_perfect_reconstruction.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_swt.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_thresholding.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_wavelet.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_wp.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test_wp2d.py -> build\lib.win-amd64-3.8\pywt\tests
    copying pywt\tests\test__pywt.py -> build\lib.win-amd64-3.8\pywt\tests
    creating build\lib.win-amd64-3.8\pywt\tests\data
    copying pywt\tests\data\cwt_matlabR2015b_result.npz -> build\lib.win-amd64-3.8\pywt\tests\data
    copying pywt\tests\data\dwt_matlabR2012a_result.npz -> build\lib.win-amd64-3.8\pywt\tests\data
    copying pywt\tests\data\generate_matlab_data.py -> build\lib.win-amd64-3.8\pywt\tests\data
    copying pywt\tests\data\generate_matlab_data_cwt.py -> build\lib.win-amd64-3.8\pywt\tests\data
    copying pywt\data\ecg.npy -> build\lib.win-amd64-3.8\pywt\data
    copying pywt\data\aero.npz -> build\lib.win-amd64-3.8\pywt\data
    copying pywt\data\ascent.npz -> build\lib.win-amd64-3.8\pywt\data
    copying pywt\data\camera.npz -> build\lib.win-amd64-3.8\pywt\data
    copying pywt\data\sst_nino3.npz -> build\lib.win-amd64-3.8\pywt\data
    running build_clib
    building 'c_wt' library
    creating build\temp.win-amd64-3.8
    creating build\temp.win-amd64-3.8\pywt
    creating build\temp.win-amd64-3.8\pywt\_extensions
    creating build\temp.win-amd64-3.8\pywt\_extensions\c
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_EXTENSION -Ipywt\_extensions\c -IC:\Users\finnk\AppData\Local\Programs\Python\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tcpywt\_extensions\c/common.c /Fobuild\temp.win-amd64-3.8\pywt\_extensions\c/common.obj
    common.c
    pywt\_extensions\c/common.c(38): warning C4244: 'return': conversion from 'unsigned long' to 'unsigned char', possible loss of data
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_EXTENSION -Ipywt\_extensions\c -IC:\Users\finnk\AppData\Local\Programs\Python\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tcpywt\_extensions\c/convolution.c /Fobuild\temp.win-amd64-3.8\pywt\_extensions\c/convolution.obj
    convolution.c
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_EXTENSION -Ipywt\_extensions\c -IC:\Users\finnk\AppData\Local\Programs\Python\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tcpywt\_extensions\c/wt.c /Fobuild\temp.win-amd64-3.8\pywt\_extensions\c/wt.obj
    wt.c
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(133): warning C4090: 'function': different 'const' qualifiers
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(134): warning C4090: 'function': different 'const' qualifiers
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(136): warning C4267: 'function': conversion from 'size_t' to 'unsigned int', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(139): warning C4090: 'function': different 'const' qualifiers
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(140): warning C4090: 'function': different 'const' qualifiers
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(142): warning C4267: 'function': conversion from 'size_t' to 'unsigned int', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(469): warning C4334: '<<': result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?)
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(133): warning C4090: 'function': different 'const' qualifiers
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(134): warning C4090: 'function': different 'const' qualifiers
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(136): warning C4267: 'function': conversion from 'size_t' to 'unsigned int', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(139): warning C4090: 'function': different 'const' qualifiers
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(140): warning C4090: 'function': different 'const' qualifiers
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(142): warning C4267: 'function': conversion from 'size_t' to 'unsigned int', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\wt.template.c(469): warning C4334: '<<': result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?)
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_EXTENSION -Ipywt\_extensions\c -IC:\Users\finnk\AppData\Local\Programs\Python\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tcpywt\_extensions\c/wavelets.c /Fobuild\temp.win-amd64-3.8\pywt\_extensions\c/wavelets.obj
    wavelets.c
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_EXTENSION -Ipywt\_extensions\c -IC:\Users\finnk\AppData\Local\Programs\Python\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tcpywt\_extensions\c/cwt.c /Fobuild\temp.win-amd64-3.8\pywt\_extensions\c/cwt.obj
    cwt.c
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(24): warning C4244: 'return': conversion from 'double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(32): warning C4244: 'return': conversion from 'double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(40): warning C4244: 'return': conversion from 'double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(48): warning C4244: 'return': conversion from 'double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(56): warning C4244: 'return': conversion from 'double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(64): warning C4305: 'return': truncation from 'double' to 'float'
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(26): warning C4244: 'function': conversion from 'const double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(26): warning C4244: 'function': conversion from 'const double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(34): warning C4244: 'function': conversion from 'const double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(42): warning C4244: 'function': conversion from 'const double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(50): warning C4244: 'function': conversion from 'const double' to 'float', possible loss of data
    C:\Users\finnk\AppData\Local\Temp\pip-install-nc9fzkly\PyWavelets\pywt\_extensions\c\cwt.template.c(58): warning C4244: 'function': conversion from 'const double' to 'float', possible loss of data
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\lib.exe build\temp.win-amd64-3.8\pywt\_extensions\c/common.obj build\temp.win-amd64-3.8\pywt\_extensions\c/convolution.obj build\temp.win-amd64-3.8\pywt\_extensions\c/wt.obj build\temp.win-amd64-3.8\pywt\_extensions\c/wavelets.obj build\temp.win-amd64-3.8\pywt\_extensions\c/cwt.obj /OUT:build\temp.win-amd64-3.8\c_wt.lib
    Microsoft (R) Library Manager Version 14.25.28612.0
    Copyright (C) Microsoft Corporation.  All rights reserved.

    running build_ext
    building 'pywt._extensions._pywt' extension
    creating build\temp.win-amd64-3.8\Release
    creating build\temp.win-amd64-3.8\Release\pywt
    creating build\temp.win-amd64-3.8\Release\pywt\_extensions
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_EXTENSION -Ipywt\_extensions\c -Ic:\users\finnk\pycharmprojects\onkodicom\venv\lib\site-packages\numpy\core\include -Ic:\users\finnk\pycharmprojects\onkodicom\venv\include -IC:\Users\finnk\AppData\Local\Programs\Python\Python38\include -IC:\Users\finnk\AppData\Local\Programs\Python\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.25.28610\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tcpywt\_extensions\_pywt.c /Fobuild\temp.win-amd64-3.8\Release\pywt\_extensions\_pywt.obj
    _pywt.c
    c:\users\finnk\pycharmprojects\onkodicom\venv\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
    pywt\_extensions\_pywt.c(35012): error C2039: 'exc_type': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35013): error C2039: 'exc_value': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35014): error C2039: 'exc_traceback': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35021): error C2039: 'exc_type': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35022): error C2039: 'exc_value': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35023): error C2039: 'exc_traceback': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35024): error C2039: 'exc_type': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35025): error C2039: 'exc_value': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35026): error C2039: 'exc_traceback': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35081): error C2039: 'exc_type': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35082): error C2039: 'exc_value': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35083): error C2039: 'exc_traceback': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35084): error C2039: 'exc_type': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35085): error C2039: 'exc_value': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(35086): error C2039: 'exc_traceback': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(36973): error C2039: 'exc_type': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(36974): error C2039: 'exc_value': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(36975): error C2039: 'exc_traceback': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(36976): error C2039: 'exc_type': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(36977): error C2039: 'exc_value': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    pywt\_extensions\_pywt.c(36978): error C2039: 'exc_traceback': is not a member of '_ts'
    C:\Users\finnk\AppData\Local\Programs\Python\Python38\include\cpython/pystate.h(51): note: see declaration of '_ts'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.25.28610\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
```

Apologies if this is a stupid question with a simple solution.
Thanks for your help.

**Comment by grlee77** [comment]
I think this may be caused by C files generated by an older Cython that bundled in the source distribution on PyPI. Are you trying to build the most recent release from there?

Can you make sure that Cython is in the virtualenv and that the version is fairly recent (I think it has to be >=  0.29.13 or so for Python 3.8)? If Cython is not trying to regenerate the C files when running 
`pip install pywavelets`
Can you try instead running directly from the github repo which does not have the Cython-generated files present?
`pip install git+https://github.com/PyWavelets/pywt.git@v1.1.1`

**Comment by Fincap** [comment]
Thanks for your quick reply.

I was trying to build the most recent release from PyPi, correct. I tried running directly from the repo, but I was getting `Module not found: Cython`
I then realized I have no Cython files in the venv either (and pip wasn't installing it). I believe I've misunderstood the installation instructions because at no point had I installed Cython. Installing Cython into the venv has worked and I have successfully installed PyWavelets. I think I've misinterpreted the installation instructions and hadn't realised that Cython was a dependency. Is Cython something that's supposed to install from the PyPi source distribution?

Thanks for your help and sorry for presenting an issue that seemed to be entirely caused my own ignorance.
